raiden/network/proxies/token_network.py

```diff
@@ -809,96 +861,178 @@ def _set_total_deposit(
+                )
+
+                network_balance = self.token.balance_of(
+                    address=Address(self.address), block_identifier=receipt["blockHash"]
+                )
+
+                if network_balance + deposit_amount > token_network_deposit_limit:
+                    msg = (
+                        f"Deposit of {deposit_amount} would have "
+                        f"exceeded the token network deposit limit."
+                    )
+                    raise RaidenRecoverableError(msg)
+
+                channel_participant_deposit_limit = self.channel_participant_deposit_limit(
+                    block_identifier=receipt["blockHash"]
+                )
+                if total_deposit > channel_participant_deposit_limit:
+                    msg = (
+                        f"Deposit of {total_deposit} is larger than the "
+                        f"channel participant deposit limit"
+                    )
+                    raise RaidenUnrecoverableError(msg)
+
+                allowance = self.token.allowance(
+                    owner=self.node_address,
+                    spender=Address(self.address),
+                    block_identifier=failed_at_blockhash,
+                )
+                has_sufficient_balance = (
+                    self.token.balance_of(self.node_address, given_block_identifier)
+                    < amount_to_deposit
+                )
+                if allowance < amount_to_deposit:
+                    msg = (
+                        f"The allowance of the {amount_to_deposit} deposit changed. "
+                        f"Check concurrent deposits "
+                        f"for the same token network but different proxies."
+                    )
+                    raise RaidenUnrecoverableError(msg)
+
+                latest_deposit = self._detail_participant(
+                    channel_identifier=channel_identifier,
+                    detail_for=self.node_address,
+                    partner=partner,
+                    block_identifier=failed_at_blockhash,
+                ).deposit
+                if latest_deposit < total_deposit:
+                    raise RaidenUnrecoverableError("The tokens were not transferred")
+
+                raise RaidenUnrecoverableError("Unlocked failed for an unknown reason")
+        else:
+            # The transaction has failed, figure out why.
+
+            # The latest block can not be used reliably because of reorgs,
+            # therefore every call using this block has to handle pruned data.
+            failed_at = self.proxy.jsonrpc_client.get_block("latest")
+            failed_at_blockhash = encode_hex(failed_at["hash"])
+            failed_at_blocknumber = failed_at["number"]
 
             self.proxy.jsonrpc_client.check_for_insufficient_eth(
                 transaction_name="setTotalDeposit",
-                transaction_executed=transaction_executed,
+                transaction_executed=True,
```

Comment:
This should be `False`, the gas estimation failed

Reply:
Done